loopy/schedule/tools.py

```diff
@@ -862,10 +863,14 @@ def _update_nesting_constraints(
                                                 .ancestors(inner_iname_nest))
                     ancestors_of_outer_iname = (loop_nest_tree
                                                 .ancestors(outer_iname_nest))
-                    if outer_iname in ancestors_of_inner_iname:
```

Comment:
Would have been nice if `mypy` could tell us our comparison types are incompatible. We were checking "`InameStr` in `tuple[InameStrSet, ...]`" which would always return False.